modules/storage-api/src/main/java/org/apache/ignite/internal/storage/ReadFromDestroyedIndexStorageException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage;
+
+import static org.apache.ignite.lang.ErrorGroups.Storage.READING_FROM_ALREADY_DESTROYED_INDEX_ERR;
+
+/**
+ * Thrown if an attempt is made to read from an index storage that is under destruction or already destroyed.
+ */
+public class ReadFromDestroyedIndexStorageException extends StorageDestroyedException {
```

Comment:
What is different between this new Exception with new code and `StorageDestroyedException` with a new error code? Can we keep just single code and single Exception here?

Reply:
There are 2 cases when a 'storage is destroyed' might be thrown, they need to be handled differently. That's why there are 2 exceptions with 2 code.

Reply:
Could you provide what will be different for handle these two situations? 
Also Index it is a storage also...

Reply:
I removed the added codes as it really seems an overkill to have them just for this rare corner case. The use will still get an error message saying that there was a read from a destroyed index storage.